projects/components/table/src/data/table-data-source.ts

```diff
@@ -275,9 +278,10 @@ export class PsTableDataSource<T> extends DataSource<T> {
 
           this.dataLength = filterResult.TotalItems;
           this.data = filterResult.Items;
-
           this._checkPageValidity(filterResult.TotalItems);
         }
+
+        this.pages = Math.round(this.dataLength / this.pageSize);

```

Comment:
I think you should round up there:
num items | pagesize | expected | Math.round result
15 | 5 | 3 | 3
14 | 5 | 3 | 3
11 | 5 | 3 | 2 <-- problem